captum/attr/_core/shapley_value.py

```diff
@@ -265,6 +265,14 @@ def attribute(
         # Keeps track whether original input is a tuple or not before
         # converting it into a tuple.
         is_inputs_tuple = _is_tuple(inputs)
+
+        # Inputs that are subtypes of float will cause errors further down.
+        # To allow such inputs we convert immediately.
+        if is_inputs_tuple:
+            inputs = tuple([t.float() for t in inputs])
+        else:
+            inputs = inputs.float()
```

Comment:
I'm not sure if we want to convert the inputs here, this assumes that the model will support taking a float tensor as input, which may not be the case for embedding inputs etc. The input used to run the model's forward pass should ideally maintain the same type as the input provided for attribution.

Reply:
Fair point; the type assumptions are a bit too strong. I'll find a workaround.

Reply:
Sounds good, thanks!